lib/ansible/modules/windows/win_rds_rap.ps1

```diff
@@ -0,0 +1,252 @@
+        $rap.PortNumbers = @("any")
+    } else {
+        $rap.PortNumbers = @($rap.PortNumbers -split ',')
+    }
+
+    # Fetch RAP user groups
+    $rap.UserGroups = @(Get-ChildItem -Path "$rap_path\UserGroups" | Select -ExpandProperty Name)
+
+    return $rap
+}
+
+function Set-RAPPropertyValue {
+    [CmdletBinding(SupportsShouldProcess=$true)]
+    param (
+        [Parameter(Mandatory=$true)]
+        [string] $name,
+        [Parameter(Mandatory=$true)]
+        [string] $property,
+        [Parameter(Mandatory=$true)]
+        $value,
+        [Parameter()]
+        $resultobj = @{}
+    )
+
+    $rap_path = "RDS:\GatewayServer\RAP\$name"
+
+    try {
+        Set-Item -Path "$rap_path\$property" -Value $value -ErrorAction stop
+    } catch {
+        Fail-Json -obj $resultobj -message "Failed to set property $property of RAP ${name}: $($_.Exception.Message)"
+    }
+}
+
+$result = @{
+  changed = $false
+}
+$diff_text = $null
+
+# Validate RAP name
+if ($name -match "[*/\\;:?`"<>|\t]+") {
+    Fail-Json -obj $result -message "Invalid character in RAP name."
+}
+
+# Validate user groups
+if ($null -ne $user_groups) {
+    if ($user_groups.Count -lt 1) {
+        Fail-Json -obj $result -message "Parameter 'user_groups' cannot be an empty list."
+    }
+
+    $user_groups = $user_groups | foreach {
+        $group = $_
+        # Test that the group is resolvable on the local machine
+        $sid = Convert-ToSID -account_name $group
+        if (!$sid) {
+            Fail-Json -obj $result -message "$group is not a valid user group on the host machine or domain"
+        }
+
+        # Return the normalized group name in UPN format
+        $group_name = Convert-FromSID -sid $sid
+        ($group_name -split "\\")[1..0] -join "@"
+    }
+    $user_groups = @($user_groups)
+}
+
+# Validate computer group parameter
+if ($computer_group_type -eq "allow_any" -and $null -ne $computer_group) {
+    Add-Warning -obj $result -message "Parameter 'computer_group' ignored because the computer_group_type is set to allow_any."
+} elseif ($computer_group_type -eq "rdg_group" -and -not (Test-Path -Path "RDS:\GatewayServer\GatewayManagedComputerGroups\$computer_group")) {
+    Fail-Json -obj $result -message "$computer_group is not a valid gateway managed computer group"
+} elseif ($computer_group_type -eq "ad_network_resource_group") {
+    $sid = Convert-ToSID -account_name $computer_group
+    if (!$sid) {
+        Fail-Json -obj $result -message "$computer_group is not a valid computer group on the host machine or domain"
+    }
+    # Ensure the group name is in UPN format
+    $computer_group = Convert-FromSID -sid $sid
+    $computer_group = ($computer_group -split "\\")[1..0] -join "@"
+}
+
+# Validate port numbers
+if ($null -ne $allowed_ports) {
+    foreach ($port in $allowed_ports) {
+        if ($port -notmatch "^([1-9][0-9]{0,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5]|any)$") {
```

Comment:
Regex is a bit strange and confusing to me, can you just verify if the int falls within a range of numbers instead?